rathole: 0.5.0 -> 0.5.0-unstable-2024-06-06

xokdvium opened the pull request:
## Description of changes

Fixes build failure introduced with Rust 1.80. See tracking issue: https://github.com/NixOS/nixpkgs/issues/332957.
The necessary time crate bump is already in main: https://github.com/rapiz1/rathole/issues/380

- Move to pkgs/by-name
- Update to [be14d12](https://github.com/rapiz1/rathole/commit/be14d12), which has Rust 1.80 fixes
- Re-enable doCheck
- Add myself as maintainer
- Sort meta attributes
- Remove with lib; usage
- Drop changelog for unstable version, since upstream does not track unreleased changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

alyssais reviewed (dismissed):
Please split this into multiple commits.

Commit:
rathole: move to pkgs/by-name

Commit:
rathole: format with nixfmt

xokdvium commented:
@ofborg eval

alyssais commented on pkgs/by-name/ra/rathole/package.nix:
```diff
@@ -30,18 +30,20 @@ rustPlatform.buildRustPackage rec {
   __darwinAllowLocalNetworking = true;
 
-  doCheck = false; # https://github.com/rapiz1/rathole/issues/222
+  doCheck = true;
```
This is the default.

alyssais reviewed (commented):
In future when fixing a broken package, it'd be better to focus on just fixing the breakage, as it takes longer to find the time to review and merge a PR that makes a lot of changes.

Commit:
rathole: 0.5.0 -> 0.5.0-unstable-2024-06-06

- Update to [be14d12](https://github.com/rapiz1/rathole/commit/be14d12),
  which has Rust 1.80 fixes
- Re-enable doCheck
- Add myself as maintainer
- Sort meta attributes
- Remove with lib; usage
- Drop changelog for unstable version, since upstream
  does not track unreleased changes

alienzj commented:
Hi, may I know can you all merge it?
